sql/parser/builtins.go

```diff
@@ -943,6 +961,125 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
+
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
```

Comment:
A global regexp cache was one of the initial ideas for the regexp-dependent expressions, and I think this is starting to show a more general need for one. I'd hate to keep running into these needs for specialized caches all over the place. It could be implemented as either a fixed size LRU cache or an expiration based cache, mapping strings to compiled `*regexp.Regexp`. I'd say an LRU cache is probably more ideal because we could put an upper bound on its memory footprint, but we still might want some kind of expiration mechanism.

Also, @petermattis, the previous example got me thinking. Would a query like `SELECT '1' LIKE changing FROM (SELECT TEXT(generate_series) AS changing FROM generate_series(0, 3)) AS t;` work with the way we currently have LIKE implemented?